src/Microsoft.CodeAnalysis.Analyzers/UnitTests/MetaAnalyzers/PreferIsKindAnalyzerTests.cs

```diff
@@ -0,0 +1,423 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.PreferIsKindAnalyzer,
+    Microsoft.CodeAnalysis.CSharp.Analyzers.MetaAnalyzers.Fixers.CSharpPreferIsKindFix>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.PreferIsKindAnalyzer,
+    Microsoft.CodeAnalysis.VisualBasic.Analyzers.MetaAnalyzers.CodeFixes.BasicPreferIsKindFix>;
+
+namespace Microsoft.CodeAnalysis.Analyzers.UnitTests.MetaAnalyzers
+{
+    public class PreferIsKindAnalyzerTests
+    {
+        [Theory]
+        [InlineData("==")]
+        [InlineData("!=")]
+        public async Task TestSimpleReturn_CS(string @operator)
+        {
+            var source =
+$@"using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+
+class C
+{{
+    bool Method(SyntaxNode node)
+    {{
+        return [|node.Kind()|] {@operator} SyntaxKind.None;
+    }}
+}}
+";
+
+            var prefix = @operator switch
```

Comment:
Nit: can have the prefix be another test method parameter. This is fine as well though..

Reply:
I'm trying out a new pattern where test _inputs_ are provided by parameters, but expected _outputs_ are provided by switch expressions.